From f6ac2de1f031994764d5bed39b79298929ef5b67 Mon Sep 17 00:00:00 2001
From: "M. Vefa Bicakci" <vefa.bicakci@windriver.com>
Date: Fri, 14 Jan 2022 17:39:52 -0500
Subject: [PATCH 1/8] iavf_main: Use irq_update_affinity_hint

This commit makes iavf_main use irq_update_affinity_hint instead of
irq_set_affinity_hint to set the CPU affinity hints. This is done
because the latter function sets the IRQ CPU affinities, whereas the
former does not, and this allows the use of the default IRQ affinity CPU
mask provided via the irqaffinity= kernel command line option.

This commit essentially replicates the iavf patch in the following
patch series:
  https://lore.kernel.org/netdev/20210903152430.244937-1-nitesh@redhat.com/t/#u

The iavf patch has been mainlined as of this writing:
  https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=0f9744f4ed539f2e847d7ed41993b243e3ba5cff

Signed-off-by: M. Vefa Bicakci <vefa.bicakci@windriver.com>
---
 src/iavf_main.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/iavf_main.c b/src/iavf_main.c
index 69059ac..cc5ca85 100644
--- a/src/iavf_main.c
+++ b/src/iavf_main.c
@@ -544,10 +544,10 @@ iavf_request_traffic_irqs(struct iavf_adapter *adapter, char *basename)
 #ifdef HAVE_IRQ_AFFINITY_HINT
 		/* Spread the IRQ affinity hints across online CPUs. Note that
 		 * get_cpu_mask returns a mask with a permanent lifetime so
-		 * it's safe to use as a hint for irq_set_affinity_hint.
+		 * it's safe to use as a hint for irq_update_affinity_hint.
 		 */
 		cpu = cpumask_local_spread(q_vector->v_idx, -1);
-		irq_set_affinity_hint(irq_num, get_cpu_mask(cpu));
+		irq_update_affinity_hint(irq_num, get_cpu_mask(cpu));
 #endif /* HAVE_IRQ_AFFINITY_HINT */
 	}
 
@@ -561,7 +561,7 @@ free_queue_irqs:
 		irq_set_affinity_notifier(irq_num, NULL);
 #endif
 #ifdef HAVE_IRQ_AFFINITY_HINT
-		irq_set_affinity_hint(irq_num, NULL);
+		irq_update_affinity_hint(irq_num, NULL);
 #endif
 		free_irq(irq_num, &adapter->q_vectors[vector]);
 	}
@@ -617,7 +617,7 @@ static void iavf_free_traffic_irqs(struct iavf_adapter *adapter)
 		irq_set_affinity_notifier(irq_num, NULL);
 #endif
 #ifdef HAVE_IRQ_AFFINITY_HINT
-		irq_set_affinity_hint(irq_num, NULL);
+		irq_update_affinity_hint(irq_num, NULL);
 #endif
 		free_irq(irq_num, &adapter->q_vectors[vector]);
 	}
-- 
2.42.0

